src/video.cpp

```diff
@@ -624,21 +624,26 @@ static void Blit32bppImages(const ClippedRectangle &cr, int32 x_base, int32 y_ba
 		for (;;) {
 			uint8 mode = *src++;
 			if (mode == 0) break;
+			const bool is_in_drawing_region = (src_base >= cr.address && src_base < cr.address + cr.width * cr.height);
 			switch (mode >> 6) {
 				case 0: // Fully opaque pixels.
 					mode &= 0x3F;
 					if (shift == GS_SEMI_TRANSPARENT) {
 						src += 3 * mode;
 						for (; mode > 0; mode--) {
-							uint32 ndest = BlendPixels(255, 255, 255, *src_base, OPACITY_SEMI_TRANSPARENT);
-							BlitPixel(cr, src_base, xpos, ypos, numx, numy, spr->width, spr->height, ndest);
+							if (is_in_drawing_region) {
```

Comment:
This condition doesn't change in the loop so it shouldn't be here, also, it doesn't do full protection as `src_base` changes here.